src/vs/workbench/contrib/markers/browser/markersTable.ts

```diff
@@ -441,7 +443,19 @@ export class MarkersTable extends Disposable implements IProblemsWidget {
 			}
 		}
 		this._itemCount = items.length;
-		this.table.splice(0, Number.POSITIVE_INFINITY, items.sort((a, b) => MarkerSeverity.compare(a.marker.severity, b.marker.severity)));
+		this.table.splice(0, Number.POSITIVE_INFINITY, items.sort((a, b) => {
+			let result = MarkerSeverity.compare(a.marker.severity, b.marker.severity);
+
+			if (result === 0) {
+				result = comparePaths(a.marker.resource.fsPath, b.marker.resource.fsPath);
```

Comment:
It's not recommended to use fsPath unless we are sure the resource is a file on disk. To compare markers by URI please use this - https://github.com/microsoft/vscode/blob/553fc81c7ecea8af02f6580a6d1a3d35941dad3a/src/vs/workbench/contrib/markers/browser/markersModel.ts#L20

Reply:
Fixed. Thanks!